README.md

```diff
@@ -1,2 +1,133 @@
+
+   Once a component is defined, we wrap them with the necessary code to
+   turn them into servers ready to take requests from Dapr. You can take a 
+   look on [StateStoreComponentServer.java] for an example of how we did that
+   for [InMemoryStateStore.java].
+
+   We also took care of all the tedious work for step 2 by defining a wrapper
+   that handles all the tedious work of handling command line parsing,
+   setting up a server listening on a Unix Socket Domain etc.
+
+TODO: talk about service discovery and how we tie a running component server to
+a component configuration as `config/container_java_memstore.yaml`.
+
+# Compiling and running the code
+
+The easiest way to get up and running is using docker-compose. 
+
+## Running using docker-compose
+
+Running using `docker-compose` allows to quickly test the whole solution with no compilation required. It also portrays how one must orchestrate the start of pluggable components and `daprd` itself.
+
+In the current directory run:
+
+```bash
+docker-compose up
+```
+
+If you rather have this ran in detached mode, add  `-d` parameter at the end of that command.
+
+You can run the commands from the Testing section bellow after starting running `docker-compose`.
+
+
+## Running as a Docker image
+
+Another option is to build a Docker image for the component and run it from there.
+
+You might want to take this route once your component is stable and is ready to
+test with your local Dapr instance.
+
+Build and start the container:
+
+```
+docker build -t javacomponent .
+mkdir -pv  /tmp/sharedUDS
+docker run -e DAPR_COMPONENT_SOCKET_PATH=/tmp/sharedUDS/javaMemstore.socket -v /tmp/sharedUDS/:/tmp/sharedUDS/ javacomponent
+```
+
+> ⚠️ Remember, you need to start your component **before** you start `daprd`.
+
+
+## Compiling from sources and running locally
+
+In the early days of development you might want to build and run the component from the command line.
+
+To test things and quickly get a server running:
+```bash
+./gradle build
+./gradlew run --args="--help"
+```
+
+If you rather a wrapper script installed so you could call this server as a
+regular application, run the following:
+
+```shell
+./gradlew installDist
+```
+
+This will create an application that can be run directly:
+
+```shell
+./build/install/DaprPluggableComponent-Java/bin/state-store-component-server --help
+```
+
+The intended way to use this application is setting an environment variable  `DAPR_COMPONENT_SOCKET_PATH` that will point to the unix domain socket file this server will listen to:
+```shell
+DAPR_COMPONENT_SOCKET_PATH=/tmp/unix.sock ./build/install/DaprPluggableComponent-Java/bin/state-store-component-server
+```
+
+> ⚠️ Remember, you need to start your component **before** you start `daprd`.
+
+# Testing
+
+Start Dapr following the build process as described in [Pluggable Components QuickStart], this time pointing components path to this project's `config` directory:
```

Comment:
should we add a link for the [Pluggable Components QuickStart]?

Reply:
No, this is a reference link. That reference is defined at the end of the document in line 131.
https://daringfireball.net/projects/markdown/syntax#link and https://www.markdownguide.org/basic-syntax/#reference-style-links